tortoise/verifying_tortoise.go

```diff
@@ -9,34 +10,22 @@ import (
 
 	"github.com/syndtr/goleveldb/leveldb"
 
+	"github.com/spacemeshos/go-spacemesh/blocks"
 	"github.com/spacemeshos/go-spacemesh/common/types"
 	"github.com/spacemeshos/go-spacemesh/database"
 	"github.com/spacemeshos/go-spacemesh/log"
 	"github.com/spacemeshos/go-spacemesh/mesh"
 	"github.com/spacemeshos/go-spacemesh/tortoise/metrics"
 )
 
-type blockDataProvider interface {
-	LayerContextuallyValidBlocks(context.Context, types.LayerID) (map[types.BlockID]struct{}, error)
-	GetBlock(types.BlockID) (*types.Block, error)
-	LayerBlockIds(types.LayerID) ([]types.BlockID, error)
-	LayerBlocks(types.LayerID) ([]*types.Block, error)
-
-	GetCoinflip(context.Context, types.LayerID) (bool, bool)
-	GetLayerInputVectorByID(types.LayerID) ([]types.BlockID, error)
-	SaveContextualValidity(types.BlockID, types.LayerID, bool) error
-	ContextualValidity(types.BlockID) (bool, error)
-}
-
 type atxDataProvider interface {
 	GetAtxHeader(types.ATXID) (*types.ActivationTxHeader, error)
 }
 
-type layerClock interface {
-	LayerToTime(types.LayerID) time.Time
-}
-
 var (
+	// we will delay counting votes in blocks with different beacon values by this many layers.
+	badBeaconVoteDelays = uint32(10)
```

Comment:
can it be a constant? or otherwise shouldn't it be defined with config?

Reply:
at this point this value is quite arbitrary. i think we need to do experiment to finalize this value. during R&D discussion Tal had always used arbitrary layers, most often 1 epoch. so i set it in the config with `layersPerEpoch`